pkg/dyninst/rcscrape/scraper.go

```diff
@@ -67,33 +78,118 @@ func NewScraper[A Actuator[T], T ActuatorTenant](
 	return v
 }
 
-// ProcessUpdate is a wrapper around an actuator.ProcessUpdate that includes
-// the runtime ID of the process.
+// ProcessUpdate represents the current state of a process, which may have
+// changed since the last time the Scraper returned information about that
+// process. An update doesn't tell you exactly what changed, and ProcessUpdates
+// can be produced even when nothing at all changed.
 type ProcessUpdate struct {
 	procmon.ProcessUpdate
 	RuntimeID         string
 	Probes            []ir.ProbeDefinition
 	ShouldUploadSymDB bool
 }
 
-// GetUpdates returns the current set of updates.
+// GetUpdates returns the current state of processes that have pending updates
+// (i.e. updates not previously returned by GetUpdates).
 func (s *Scraper) GetUpdates() []ProcessUpdate {
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	return s.mu.debouncer.coalesceInFlight(time.Now())
+	updates := s.mu.debouncer.getUpdates(time.Now())
+	res := make([]ProcessUpdate, 0, len(updates))
+	for procID, u := range updates {
+		p, ok := s.mu.processes[procID]
+		if !ok {
+			log.Warnf("bug: debouncer update for untracked process: %v", procID)
```

Comment:
I don't think this is definitely a bug because there isn't synchronization between the removal of the probes and the processing of the ringbuf, but, if this keeps happening, then it's a bug. I sort of liked that about the old implementation, but whatever.